cloudify_agent/tests/installer/config/__init__.py

```diff
@@ -39,5 +39,6 @@ def mock_context(agent_properties=None,
         bootstrap_context=BootstrapContext(
             bootstrap_context={'cloudify_agent': agent_context})
     )
-    setattr(context, 'installer', MagicMock())
+    context.installer = MagicMock()
+    context._get_current_object = lambda: context
```

Comment:
btw `_get_current_object` is actually public api; pocoo projects commonly use _ to prefix attributes to avoid name conflicts, if the object is allowed to have a user-provided attribute named the same: http://werkzeug.pocoo.org/docs/0.11/local/#werkzeug.local.LocalProxy._get_current_object
(we use `proxy_tools` which is copied from werkzeug)